### DISCUSSION TITLE
Obtain Ref from main.dart before flutter initialization

### DISCUSSION THREAD

**Discussion by martinc2907** [rrousselGit/riverpod#1502] [Q&A] [answered]
Is there a way to obtain the ref object in main.dart before using the ProviderScope? My code looks like the following, and my repository class requires the ref object. The reason for initializing my repository in this fashion is because it requires an async initialization method. 

```
main(){
  final repository = Repository(//NEED REF HERE);
  await repository.init();

  runApp(
    ProviderScope(overrides:[ repositoryProvider.overrideWithValue(repository) ], child: App())
  );
}
```

**Comment by tbm98** [comment, marked as answer]
[riverpod-initialize-listener-app-startup](https://codewithandrea.com/articles/riverpod-initialize-listener-app-startup/#conclusion) may help you.
